File: app/src/main/java/dagger/nambv/app/demo/di/component/AppComponent.java
package dagger.nambv.app.demo.di.component;

import javax.inject.Singleton;

import dagger.Component;
import dagger.nambv.app.demo.di.module.AppModule;
import dagger.nambv.app.demo.di.module.CommonModule;
import dagger.nambv.app.demo.view.fragment.PromotionsFragmentComponent;
import dagger.nambv.app.demo.view.fragment.PromotionsFragmentModule;

@Singleton
@Component(modules = {AppModule.class, CommonModule.class})
public interface AppComponent {
    PromotionsFragmentComponent plus(PromotionsFragmentModule module);
}

File: app/src/main/java/dagger/nambv/app/demo/di/component/AppComponent.kt
package dagger.nambv.app.demo.di.component

import dagger.Component
import dagger.nambv.app.demo.di.module.AppModule
import dagger.nambv.app.demo.di.module.CommonModule
import dagger.nambv.app.demo.view.fragment.PromotionsFragmentComponent
import dagger.nambv.app.demo.view.fragment.PromotionsFragmentModule
import javax.inject.Singleton

@Singleton
@Component(modules = arrayOf(AppModule::class, CommonModule::class))
interface AppComponent {
    operator fun plus(module: PromotionsFragmentModule): PromotionsFragmentComponent
}
